File: leetcode3/변지협/v3/682. Baseball Game.py
'''
1. 아이디어:
문제에 주어진 대로 풀이. 스택 사용
2. 시간복잡도:
o(n)
3. 알고리즘:
스택
'''

class Solution:
    def calPoints(self, operations: List[str]) -> int:
        stack = []
        for op in operations:
            if op == '+':
                stack.append(stack[-1] + stack[-2])
            elif op == 'D':
                stack.append(stack[-1] * 2)
            elif op == 'C':
                stack.pop()
            else:
                stack.append(int(op))

        return sum(stack)
        

File: leetcode3/황은지/682. Baseball Game.js
/**
 * @param {string[]} operations
 * @return {number}
 */
var calPoints = function (operations) {
  const stack = [];
  for (const op of operations) {
    const len = stack.length;
    if (op === "D") {
      const x = stack[len - 1];
      stack.push(+x * 2);
    } else if (op === "+") {
      const x = stack[len - 1];
      const y = stack[len - 2];
      stack.push(+x + +y);
    } else if (op === "C") stack.pop();
    else stack.push(+op);

    console.log(stack);
  }

  let result = 0;
  for (const num of stack) {
    result += num;
  }

  return result;
};
